e2etest/l2tests/l2.go

```diff
@@ -153,15 +153,22 @@ var _ = ginkgo.Describe("L2", func() {
 
 			ginkgo.By("validating the node in ServiceL2Status is the one who is announcing for the service")
 			l2Status := l2Statuses[0]
-			refreshL2StatusNodeFunc := func() string {
+			var speakingNode string
+			Eventually(func() string {
 				node, err := k8s.GetSvcNode(cs, svc.Namespace, svc.Name, allNodes)
 				if err != nil {
 					return err.Error()
 				}
+				speakingNode = node.Name
 				return node.Name
-			}
-			Eventually(refreshL2StatusNodeFunc, time.Minute, time.Second).Should(Equal(l2Status.Status.Node))
-			Consistently(refreshL2StatusNodeFunc, 5*time.Second).Should(Equal(l2Status.Status.Node))
+			}, time.Minute, time.Second).Should(Equal(l2Status.Status.Node))
+			Consistently(func() string {
+				var ss []metallbv1beta1.ServiceL2Status
+				if ss, err = status.GetSvcL2Status(ConfigUpdater.Client(), svc); err != nil || len(ss) != 1 {
+					return ""
+				}
+				return ss[0].Status.Node
+			}, 5*time.Second).Should(Equal(speakingNode))
 		})
```

Comment:
can we also add validation that the status is deleted after the service is deleted? (I think it's not there, ignore this if I'm wrong)